File: templates/Uwp/Features/ShareTarget.Prism/ViewModels/SharedDataStorageItemsViewModel.cs
using System;
using System.Collections.ObjectModel;
using System.Threading.Tasks;

using Windows.ApplicationModel.DataTransfer;
using Windows.ApplicationModel.DataTransfer.ShareTarget;
using Windows.Storage;
using Windows.UI.Xaml.Media;
using Windows.UI.Xaml.Media.Imaging;

using Param_RootNamespace.Helpers;

namespace Param_RootNamespace.ViewModels
{
    public class SharedDataStorageItemsViewModel : SharedDataViewModelBase
    {
        public ObservableCollection<ImageSource> Images { get; } = new ObservableCollection<ImageSource>();

        public SharedDataStorageItemsViewModel()
        {
        }

        public override async Task LoadDataAsync(ShareOperation shareOperation)
        {
            await base.LoadDataAsync(shareOperation);

            PageTitle = "ShareTargetFeature_ImagesTitle".GetLocalized();
            DataFormat = StandardDataFormats.StorageItems;
            var files = await shareOperation.GetStorageItemsAsync();
            foreach (var file in files)
            {
                var storageFile = file as StorageFile;
                if (storageFile != null)
                {
                    using (var inputStream = await storageFile.OpenReadAsync())
                    {
                        var img = new BitmapImage();
                        img.SetSource(inputStream);
                        Images.Add(img);
                    }
                }
            }
        }
    }
}

File: code/TemplateStudioForUWP/Templates/Uwp/Features/ShareTarget._VB/ViewModels/SharedDataStorageItemsViewModel.vb
Imports Windows.ApplicationModel.DataTransfer
Imports Windows.ApplicationModel.DataTransfer.ShareTarget
Imports Windows.Storage

Imports Param_RootNamespace.Helpers

Namespace ViewModels
    Public Class SharedDataStorageItemsViewModel
        Inherits SharedDataViewModelBase

        Public ReadOnly Property Images As ObservableCollection(Of ImageSource) = new ObservableCollection(Of ImageSource)

        Public Sub New()
        End Sub

        Public Overrides Async Function LoadDataAsync(shareOperation As ShareOperation) As Task
            Await MyBase.LoadDataAsync(shareOperation)

            PageTitle = "ShareTargetFeature_ImagesTitle".GetLocalized()
            DataFormat = StandardDataFormats.StorageItems
            Dim files = Await shareOperation.GetStorageItemsAsync()
            For Each file As IStorageFile In files
                Dim storageFile = TryCast(file, StorageFile)
                If storageFile IsNot Nothing Then
                    Using inputStream = Await storageFile.OpenReadAsync()
                        Dim img = New BitmapImage()
                        img.SetSource(inputStream)
                        Images.Add(img)
                    End Using
                End If
            Next
        End Function
    End Class
End Namespace
